agent/replacecall/src/main/java/randoop/instrument/MethodSignature.java

```diff
@@ -205,55 +202,51 @@ String getName() {
   }
 
   /**
-   * Uses reflection to get the {@code java.lang.reflect.Method} object for this {@link
-   * MethodSignature}.
+   * Searches the class file 'classname' for the method 'name' to get the {@code
+   * org.apache.bcel.classfile.Method} object for this {@link MethodSignature}.
    *
-   * @return the reflection object for this {@link MethodSignature}
+   * @return the Method object for this {@link MethodSignature}
    * @throws ClassNotFoundException if the containing class of this {@link MethodSignature} is not
    *     found on the classpath
    * @throws NoSuchMethodException if the containing class of this {@link MethodSignature} does not
    *     have the represented method as a member
+   * @throws IllegalClassFormatException if the containing class of this {@link MethodSignature}
+   *     exists, but cannot be loaded
    */
-  Method toMethod() throws ClassNotFoundException, NoSuchMethodException {
+  Method toMethod()
+      throws ClassNotFoundException, NoSuchMethodException, IllegalClassFormatException {
     if (method != null) {
       return method;
     }
 
-    Class<?> methodClass = Class.forName(classname);
-    Class<?> params[] = new Class[paramTypes.length];
-    for (int i = 0; i < paramTypes.length; i++) {
-      params[i] = typeToClass(paramTypes[i]);
-    }
-
-    // Note that Method.getMethod only returns public methods, so call Method.getDeclaredMethod
-    // first
+    String classname = this.classname;
+    int superclassIndex = 0;
```

Comment:
Are you saying that:
flag = (superclassIndex != 0);
while (flag); 
is preferable to:
while (superclassIndex != 0);
I think I am confused.

Reply:
If the variable was clearly named (please don't ever name a variable `flag`), then it might be better.
The main problem is that I don't understand what this condition means.  Does it mean that the current class (actually, the parent class by the end of the loop) is Object.class?  Explaining that would help.